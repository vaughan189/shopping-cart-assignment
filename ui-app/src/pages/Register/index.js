import React from "react";
import {
  Button,
  Container,
  FormControl,
  FormHelperText,
  Grid,
  Input,
  InputLabel,
  Typography,
} from "@mui/material";

import { useForm } from "react-hook-form";
import {
  emailRegexPattern,
  passwordRegexPattern,
} from "../../constants/validation";
import { useHistory } from "react-router-dom";

const Register = () => {
  let history = useHistory();
  const {
    register,
    handleSubmit,
    getValues,
    formState: { errors },
  } = useForm();

  const onSubmit = () => {
    history.push("/home");
  };

  return (
    <Container maxWidth="xl">
      <Grid container direction="row" spacing={10} sx={{ marginTop: "0%" }}>
        <Grid item xs="auto" sm="auto" md={2} xl={2}></Grid>
        
        <Grid item xs={12} sm={12} md={4} xl={4}>
          <Grid
            container
            direction="row"
            justifyContent="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <Typography
                sx={{
                  fontFamily: "Dosis",
                  fontSize: 28,
                  fontWeight: 600,
                  justifyContent: "centre",
                }}
              >
                Signup
              </Typography>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <Typography
                sx={{
                  fontFamily: "Dosis",
                  fontSize: 18,
                  fontWeight: 400,
                }}
              >
                We do not share your personal details with anyone
              </Typography>
            </Grid>
          </Grid>
        </Grid>
        
        <Grid item xs={12} sm={12} md={4} xl={4}>
          <Grid
            container
            direction="row"
            justifyContent="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="firstName">First Name</InputLabel>
                <Input
                  id="firstName"
                  type="text"
                  {...register("firstName", {
                    required: true,
                  })}
                />
                {errors.firstName && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    Please enter a first Name
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="lastName">Last Name</InputLabel>
                <Input
                  id="lastName"
                  type="text"
                  {...register("lastName", {
                    required: true,
                  })}
                />
                {errors.lastName && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    Please enter a last name
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="email">Email</InputLabel>
                <Input
                  id="email"
                  type="text"
                  {...register("email", {
                    required: true,
                    pattern: emailRegexPattern,
                  })}
                />
                {errors.email && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    Please enter a valid the email
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="password">Password</InputLabel>
                <Input
                  id="password"
                  type="password"
                  {...register("password", {
                    required: "Please enter a valid password",
                    minLength: {
                      value: 6,
                      message: "Password must have at least 6 characters",
                    },
                    pattern: passwordRegexPattern,
                  })}
                />
                {errors.password && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    {errors.password.message}
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="confirmPassword">
                  Confirm Password
                </InputLabel>
                <Input
                  id="confirmPassword"
                  type="password"
                  {...register("confirmPassword", {
                    required: "Please enter a valid password",
                    minLength: {
                      value: 6,
                      message: "Password must have at least 6 characters",
                    },
                    pattern: passwordRegexPattern,
                    validate: (value) => {
                      const { password } = getValues();
                      return (password === value) | "Passwords should match!";
                    },
                  })}
                />
                {errors.confirmPassword && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    {errors.confirmPassword.message}
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} md={12}>
              <Button
                id="submit-btn"
                variant="contained"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                  fontFamily: "Dosis",
                  textTransform: "none",
                }}
                onClick={handleSubmit(onSubmit)}
                disabled={errors.email || errors.password}
              >
                Login
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Container>
  );
};

export default Register;
