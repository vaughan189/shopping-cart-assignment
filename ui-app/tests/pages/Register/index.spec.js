import React from "react";
import reactRouterDom from "react-router-dom";
import { shallow } from "enzyme";
import Register from "../../../src/pages/Register";

jest.mock("react-router-dom");
const pushMock = jest.fn();
reactRouterDom.useHistory = jest.fn().mockReturnValue({ push: pushMock });

describe("Register Page", () => {
  it("should render the Register Page correctly", () => {
    const component = shallow(<Register debug />);
    expect(component).toMatchSnapshot();
  });

  // it("should trigger the submit function on button click", () => {
  //   const component = shallow(<Register debug />);
  //   component.find("#submit-btn").simulate("click");
  //   expect(pushMock).toHaveBeenCalled();
  // });
});
