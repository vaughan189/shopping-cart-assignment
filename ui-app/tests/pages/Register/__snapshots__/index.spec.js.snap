// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should render the Register Page correctly 1`] = `
<ForwardRef(Container)
  maxWidth="xl"
>
  <ForwardRef(Grid)
    container={true}
    direction="row"
    spacing={10}
    sx={
      Object {
        "marginTop": "0%",
      }
    }
  >
    <ForwardRef(Grid)
      item={true}
      md={2}
      sm="auto"
      xl={2}
      xs="auto"
    />
    <ForwardRef(Grid)
      item={true}
      md={4}
      sm={12}
      xl={4}
      xs={12}
    >
      <ForwardRef(Grid)
        alignItems="flex-start"
        container={true}
        direction="row"
        justifyContent="flex-start"
      >
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(Typography)
            sx={
              Object {
                "fontFamily": "Dosis",
                "fontSize": 28,
                "fontWeight": 600,
                "justifyContent": "centre",
              }
            }
          >
            Signup
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(Typography)
            sx={
              Object {
                "fontFamily": "Dosis",
                "fontSize": 18,
                "fontWeight": 400,
              }
            }
          >
            We do not share your personal details with anyone
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      md={4}
      sm={12}
      xl={4}
      xs={12}
    >
      <ForwardRef(Grid)
        alignItems="flex-start"
        container={true}
        direction="row"
        justifyContent="flex-start"
      >
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="firstName"
            >
              First Name
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="firstName"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="lastName"
            >
              Last Name
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="lastName"
              name="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="email"
            >
              Email
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="password"
            >
              Password
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="confirmPassword"
            >
              Confirm Password
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="confirmPassword"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          xs={12}
        >
          <ForwardRef(Button)
            id="submit-btn"
            onClick={[Function]}
            sx={
              Object {
                "fontFamily": "Dosis",
                "margin": 2,
                "textTransform": "none",
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="contained"
          >
            Login
          </ForwardRef(Button)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</ForwardRef(Container)>
`;
